### PR TITLE
Compile Programs Sass prior to running r.js and cache static assets

### PR DESCRIPTION
**Pull request by rlucioni**
Allows r.js to minify CSS hosted by the Programs service. This in turn yields minified files with unhashed names we can use to fetch these assets from Studio. ECOM-2862.

Corresponds to https://github.com/edx/programs/pull/69. @jimabramson please review. I'd also appreciate a look from someone in @edx/devops. @AlasdairSwan FYI.

### CODE REVIEW

**Comment by rlucioni** [comment]
I've modified Programs' Nginx configuration to allow static assets to be held in public caches for an hour.

At the moment, the only static assets served by Programs are the JS and CSS required for the Programs authoring app, an internal tool for configuring XSeries. Given that, an hour's lag between deployment and when users may actually see changes to the app is acceptable. Programs CSS and JS are requested at the same time, so I expect these to remain in sync when cached.

@feanil, since we discussed this offline today, I'd appreciate a review from you.

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/programs.j2 [hunk 1]
```diff
@@ -35,6 +35,15 @@ server {
   location ~ ^/static/(?P<file>.*) {
     root {{ COMMON_DATA_DIR }}/{{ programs_service_name }};
     try_files /staticfiles/$file =404;
+
+    # Request that the browser use SSL for these connections. Repeated here
+    # beacuse add_header directives are only inherited from the previous level
```
Comment (jimabramson) [inline playbooks/roles/nginx/templates/edx/app/nginx/sites-available/programs.j2]: typo (because)

**Comment by feanil** [comment]
:+1:  once the typo is fixed.

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/programs.j2 (hunk 1 above)
Comment (rlucioni) [inline playbooks/roles/nginx/templates/edx/app/nginx/sites-available/programs.j2]: Fixed.

**Comment by jimabramson** [comment]
:+1:

**Pull request merged**